src/ucp/core/ucp_request.c

```diff
@@ -37,6 +37,16 @@ ucs_status_t ucp_request_test(void *request, ucp_tag_recv_info_t *info)
 void ucp_request_release(void *request)
 {
     ucp_request_t *req = (ucp_request_t*)request - 1;
+    ucs_mpool_elem_t *elem = (ucs_mpool_elem_t*)req - 1;
```

Comment:
we should rely on mpool internals to get pointer to worker
IMHO we should add new API function - ucp_worker_request_release(worker, request) - and deprecate this one